metrics/tags.go

```diff
@@ -156,6 +156,82 @@ var _ interface {
 	json.Unmarshaler
 } = &TagSet{}
 
+// TagsAndMeta is a helper type that provides easy group manipulation of the
+// indexed Tags and the non-indexed Metadata values together. While both of them
+// are part of a metric Sample, the TagsAndMeta type isn't used there because
+// the Tags participate in the Sample's TimeSeries and the Metadata does not.
+//
+// Instead, this type is mostly used by the VU and JS code while it assembles
+// the final consolidated Tags and Metadata values to put in the Sample.
+//
+// IMPORTANT: this data structure is not thread safe! The methods of the
+// lib.VUStateTags type are where the synchronization happens.
+type TagsAndMeta struct {
+	Tags     *TagSet
+	Metadata map[string]string // could be nil
+}
+
+// SetTag adds the given key=value tag to the Tags and removes it from the
+// Metadata, if it was present there.
+func (tm *TagsAndMeta) SetTag(key, value string) {
+	tm.Tags = tm.Tags.With(key, value)
+	if tm.Metadata != nil {
+		delete(tm.Metadata, key)
+	}
```

Comment:
This (along with the comment above) probably needs to be removed. It doesn't make sense if we treat tags and metadata as completely different entities and not 2 flavors of the same thing.